### PR TITLE
updating all hrvo instances in the codebase

### PR DESCRIPTION
**Pull request by sarphion**
### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [x] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue

### CODE REVIEW

File: src/extlibs/hrvo/agent.h [hunk 1]
```diff
@@ -5,10 +5,10 @@
 #include "software/ai/navigator/path_planner/hrvo/velocity_obstacle.h"
 #include "software/geom/vector.h"
 
-class HRVOSimulator;
+class HrvoSimulator;
 
 /**
- * An agent/robot in the HRVO simulation.
+ * An agent/robot in the Hrvo simulation.
```
Comment (raymond212) [inline src/extlibs/hrvo/agent.h]: I think acronyms can be left in all-caps in the comments? This also applies to other changes to comments. 
```suggestion
 * An agent/robot in the HRVO simulation.
```

**Review by raymond212 — CHANGES_REQUESTED** [review]
Thanks for renaming the many HRVO instances in the codebase! I think the changes look pretty good overall. However, I believe instances of `HRVO` in comments can be left in all caps. Also, it would be helpful to link the related issue in the PR description. 

I've also noticed that it's been a few days, but the github checks haven't completed yet... Maybe try re-running the checks?

File: src/software/constants.h [hunk 2]
```diff
@@ -26,8 +26,8 @@ const std::string YELLOW_ROBOT_STATUS_PATH               = "/yellow_robot_status
 const std::string SIMULATION_TICK_PATH                   = "/simulation_tick";
 const std::string YELLOW_WORLD_PATH                      = "/yellow_world";
 const std::string BLUE_WORLD_PATH                        = "/blue_world";
-const std::string YELLOW_HRVO_PATH                       = "/yellow_hrvo";
-const std::string BLUE_HRVO_PATH                         = "/blue_hrvo";
+const std::string YELLOW_Hrvo_PATH                       = "/yellow_hrvo";
+const std::string BLUE_Hrvo_PATH                         = "/blue_hrvo";
```
Comment (williamckha) [inline src/software/constants.h]: I believe we should keep these constants in all caps

**Review by williamckha — COMMENTED** [review]
LGTM! Just a small nit

File: src/extlibs/hrvo/kd_tree.cpp [hunk 3]
```diff
@@ -89,7 +89,7 @@ void KdTree::buildRecursive(std::size_t begin, std::size_t end, std::size_t node
         }
     }
 
-    if (end - begin > HRVO_MAX_LEAF_SIZE)
+    if (end - begin > Hrvo_MAX_LEAF_SIZE)
```
Comment (AdenC123) [inline src/extlibs/hrvo/kd_tree.cpp]: constants should probably be left in all caps?

File: src/extlibs/hrvo/simulator.cpp [hunk 4]
```diff
@@ -347,22 +347,22 @@ void HRVOSimulator::visualize(unsigned int robot_id) const
 
     if (friendly_team_colour == TeamColour::YELLOW)
     {
-        LOG(VISUALIZE, YELLOW_HRVO_PATH) << hrvo_visualization;
+        LOG(VISUALIZE, YELLOW_Hrvo_PATH) << hrvo_visualization;
```
Comment (AdenC123) [inline src/extlibs/hrvo/simulator.cpp]: these constants as well

**Review by AdenC123 — COMMENTED** [review]
Same as @williamckha - believe constants should be in all caps. Otherwise looks great

File: src/extlibs/hrvo/simulated_hrvo_test.cpp [hunk 5]
```diff
@@ -300,9 +300,13 @@ TEST_F(SimulatedHRVOTest, test_agent_not_going_in_static_obstacles)
             Duration::fromSeconds(20));
 }
 
+<<<<<<< HEAD
+TEST_F(SimulatedHrvoTest, test_start_in_local_minima)
+=======
 // This test is mainly for showcasing the behaviour of HRVO in an edge case where
 // the robot starts in a local minima, as a result, it does not have any validations.
 TEST_F(SimulatedHRVOTest, test_start_in_local_minima)
+>>>>>>> c070809816e7711d3e1d65d4a1aef0a54d866bb7
```
Comment (itsarune) [inline src/extlibs/hrvo/simulated_hrvo_test.cpp]: merge conflict market, you can delete this line

File: src/extlibs/hrvo/simulated_hrvo_test.cpp [hunk 6]
```diff
@@ -300,9 +300,13 @@ TEST_F(SimulatedHRVOTest, test_agent_not_going_in_static_obstacles)
             Duration::fromSeconds(20));
 }
 
+<<<<<<< HEAD
```
Comment (itsarune) [inline src/extlibs/hrvo/simulated_hrvo_test.cpp]: git conflict marker, you may delete this line

File: src/extlibs/hrvo/simulated_hrvo_test.cpp [hunk 7]
```diff
@@ -300,9 +300,13 @@ TEST_F(SimulatedHRVOTest, test_agent_not_going_in_static_obstacles)
...
+TEST_F(SimulatedHrvoTest, test_start_in_local_minima)
+=======
```
Comment (itsarune) [inline src/extlibs/hrvo/simulated_hrvo_test.cpp]: git conflict market,  you may delete this line

File: src/extlibs/hrvo/hrvo_agent.cpp [hunk 8]
```diff
@@ -1,6 +1,6 @@
 /*
  * agent.cpp
- * HRVO Library
+ * Hrvo Library
```
Comment (itsarune) [inline src/extlibs/hrvo/hrvo_agent.cpp]: this can be HRVO (in comments)

File: src/extlibs/hrvo/hrvo_agent.cpp [hunk 9]
```diff
@@ -181,17 +181,17 @@ void HRVOAgent::computeVelocityObstacles()
...
     Circle moving_agent_circle(Point(other_agent.getPosition()), radius_);
     auto vo = generateVelocityObstacle(obstacle_agent_circle, moving_agent_circle,
                                        getVelocity());
 
-    // Convert velocity obstacle to hybrid reciprocal velocity obstacle (HRVO)
+    // Convert velocity obstacle to hybrid reciprocal velocity obstacle (Hrvo)
     // by shifting one side of the velocity obstacle to share the responsibility
     // of avoiding collision with other agent. This assumes that the other agent will also
-    // be running HRVO
+    // be running Hrvo
```
Comment (itsarune) [inline src/extlibs/hrvo/hrvo_agent.cpp]: can be HRVO here

File: src/extlibs/hrvo/hrvo_agent.h [hunk 10]
```diff
@@ -1,6 +1,6 @@
 /*
  * agent.h
- * HRVO Library
+ * Hrvo Library
```
Comment (itsarune) [inline src/extlibs/hrvo/hrvo_agent.h:3]: can be HRVO

File: src/extlibs/hrvo/hrvo_agent.h [hunk 11]
```diff
@@ -44,10 +44,10 @@
 #include "software/geom/vector.h"
 
 /**
- * An agent/robot in the simulation which uses the HRVO algorithm to motion plan towards
+ * An agent/robot in the simulation which uses the Hrvo algorithm to motion plan towards
```
Comment (itsarune) [inline src/extlibs/hrvo/hrvo_agent.h]: comments can be HRVO

File: src/extlibs/hrvo/kd_tree.cpp [hunk 12]
```diff
@@ -1,6 +1,6 @@
 /*
  * kd_tree.cpp
- * HRVO Library
+ * Hrvo Library
```
Comment (itsarune) [inline src/extlibs/hrvo/kd_tree.cpp]: comments, can be HRVO

File: src/extlibs/hrvo/kd_tree.h [hunk 13]
```diff
@@ -1,6 +1,6 @@
 /*
  * kd_tree.h
- * HRVO Library
+ * Hrvo Library
```
Comment (itsarune) [inline src/extlibs/hrvo/kd_tree.h]: comments, can be HRVO

File: src/extlibs/hrvo/simulator.cpp [hunk 14]
```diff
@@ -1,6 +1,6 @@
 /*
  * simulator.cpp
- * HRVO Library
+ * Hrvo Library
```
Comment (itsarune) [inline src/extlibs/hrvo/simulator.cpp]: comments: can be HRVO

File: src/extlibs/hrvo/simulator.h [hunk 15]
```diff
@@ -1,6 +1,6 @@
 /*
  * simulator.h
- * HRVO Library
+ * Hrvo Library
```
Comment (itsarune) [inline src/extlibs/hrvo/simulator.h]: comments: can be HRVO

File: src/software/jetson_nano/primitive_executor.cpp [hunk 16]
```diff
@@ -96,8 +96,13 @@ std::unique_ptr<TbotsProto::DirectControlPrimitive> PrimitiveExecutor::stepPrimi
 {
     hrvo_simulator_.doStep();
 
+<<<<<<< HEAD
+    // Visualize the Hrvo Simulator for the current robot
+    hrvo_simulator_.visualize(robot_id);
+=======
```
Comment (itsarune) [inline src/software/jetson_nano/primitive_executor.cpp]: conflict marker, you can delete this line

File: src/software/jetson_nano/primitive_executor.cpp [hunk 17]
```diff
@@ -96,8 +96,13 @@ std::unique_ptr<TbotsProto::DirectControlPrimitive> PrimitiveExecutor::stepPrimi
...
     // Visualize the HRVO Simulator for the current robot
     hrvo_simulator_.visualize(robot_id_);
+>>>>>>> c070809816e7711d3e1d65d4a1aef0a54d866bb7
```
Comment (itsarune) [inline src/software/jetson_nano/primitive_executor.cpp]: conflict marker, delete this line

File: src/software/jetson_nano/primitive_executor.h [hunk 18]
```diff
@@ -85,8 +85,12 @@ class PrimitiveExecutor
 
     TbotsProto::Primitive current_primitive_;
     RobotConstants_t robot_constants_;
+<<<<<<< HEAD
+    HrvoSimulator hrvo_simulator_;
+=======
     HRVOSimulator hrvo_simulator_;
```
Comment (itsarune) [inline src/software/jetson_nano/primitive_executor.h]: conflict marker: delete these lines

File: src/extlibs/hrvo/simulator.h [hunk 19]
```diff
@@ -146,11 +146,11 @@ class HRVOSimulator
     Vector getRobotVelocity(unsigned int robot_id) const;
 
     /**
-     * Get friendly HRVO agent from robot id. Returns std::nullopt if agent does not exist
+     * Get friendly Hrvo agent from robot id. Returns std::nullopt if agent does not exist
```
Comment (itsarune) [inline src/extlibs/hrvo/simulator.h]: in comments, we can use HRVO

File: src/extlibs/hrvo/simulator.h [hunk 20]
```diff
@@ -146,11 +146,11 @@ class HRVOSimulator
...
      * @param robot_id Robot id as represented in the real world
-     * @return Optional shared pointer to an HRVO agent
+     * @return Optional shared pointer to an Hrvo agent
```
Comment (itsarune) [inline src/extlibs/hrvo/simulator.h]: in comments, we can use "HRVO"

File: src/extlibs/hrvo/simulator.cpp [hunk 21]
```diff
@@ -319,15 +322,16 @@ void HRVOSimulator::updateRobotVelocity(RobotId robot_id, const Vector &new_velo
 }
 
 void HRVOSimulator::visualize(unsigned int robot_id) const
+>>>>>>> c070809816e7711d3e1d65d4a1aef0a54d866bb7
 {
     auto friendly_agent_opt = getFriendlyAgentFromRobotId(robot_id);
     if (!friendly_agent_opt.has_value())
     {
-        // HRVO friendly agent with robot id can not be visualized
+        // Hrvo friendly agent with robot id can not be visualized
```
Comment (itsarune) [inline src/extlibs/hrvo/simulator.cpp:326]: comments: "HRVO"

**Review by itsarune — COMMENTED** [review]
great start!

two notes:
- in comments, we should be using the human-readable acronym HRVO
- you have merge conflict markers, "<<<<<" and "HEAD" messages. this stops our code from compiling and we need to resolve them. you should be able to resolve them in your IDE or we can take a look at it together on Saturday

File: src/extlibs/hrvo/simulator.h [hunk 22]
```diff
@@ -316,7 +316,7 @@ class HRVOSimulator
     // NOTE: This value must be >= 0
     static constexpr float BALL_AGENT_RADIUS_OFFSET = 0.1f;
 
-    // The maximum distance which HRVO Agents will look for neighbors, in meters.
+    // The maximum distance which Hrvo Agents will look for neighbors, in meters.
```
Comment (raymond212) [inline src/extlibs/hrvo/simulator.h]: Can be HRVO

File: src/extlibs/hrvo/simulator.cpp [hunk 23]
```diff
@@ -296,20 +296,20 @@ void HRVOSimulator::doStep()
...
+Vector HrvoSimulator::getRobotVelocity(unsigned int robot_id) const
 {
     auto hrvo_agent = getFriendlyAgentFromRobotId(robot_id);
     if (hrvo_agent.has_value())
     {
         return hrvo_agent.value()->getVelocity();
     }
     LOG(WARNING) << "Velocity for robot " << robot_id
-                 << " can not be found since it does not exist in HRVO Simulator"
+                 << " can not be found since it does not exist in Hrvo Simulator"
```
Comment (raymond212) [inline src/extlibs/hrvo/simulator.cpp]: HRVO is probably okay for log messages.

**Review by raymond212 — COMMENTED** [review]
Looks pretty good, I just had two suggestions.

File: src/extlibs/hrvo/hrvo_agent.cpp [hunk 24]
```diff
@@ -181,14 +181,14 @@ void HRVOAgent::computeVelocityObstacles()
...
-VelocityObstacle HRVOAgent::createVelocityObstacle(const Agent &other_agent)
+VelocityObstacle HrvoAgent::createVelocityObstacle(const Agent &other_agent)
 {
     Circle obstacle_agent_circle(Point(getPosition()), radius_);
     Circle moving_agent_circle(Point(other_agent.getPosition()), radius_);
     auto vo = generateVelocityObstacle(obstacle_agent_circle, moving_agent_circle,
                                        getVelocity());
 
-    // Convert velocity obstacle to hybrid reciprocal velocity obstacle (HRVO)
+    // Convert velocity obstacle to hybrid reciprocal velocity obstacle (Hrvo)
```
Comment (itsarune) [inline src/extlibs/hrvo/hrvo_agent.cpp:191]:
```suggestion
    // Convert velocity obstacle to hybrid reciprocal velocity obstacle (HRVO)
```

File: src/extlibs/hrvo/simulator.cpp (hunk 21 above)
Comment (itsarune) [reply src/extlibs/hrvo/simulator.cpp:326]: bump

**Review by itsarune — COMMENTED** [review]
two small things

**Comment by nimazareian** [comment]
Closing in favour of a full HRVO rewrite in #2814
